pages/products/index.js

```diff
@@ -62,46 +63,59 @@ export default function Products() {
   const [products, setProducts] = useAtom(initialProducts);
   const router = useRouter();
 
+  const { data: storedProducts, mutate: changeProducts } =
+    useSWR(`/api/products`);
+
+  async function handleDeleteProduct(id) {
+    try {
+      await fetch(`/api/products/${id}`, {
+        method: "DELETE",
+      });
+      changeProducts();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+  useEffect(() => {
+    setProducts(storedProducts);
+  }, [storedProducts, setProducts]);
+
```

Comment:
seeing a pattern like this always leads me to linking this:

https://beta.reactjs.org/learn/you-might-not-need-an-effect

Take a look at it and maybe reevaluate if you really need that useEffect.